File: frontend/src/types/job.ts
export interface Job {
  id: string;
  title: string;
  company: string;
  companyId: string;
  sourceId: string;
  dedupKey: string;
  location: string | null;
  salary: string | null;
  description: string | null;
  applyUrl: string;
  sector: string | null;
  stage: string | null;
  isNew: boolean;
  isRemote: boolean;
  savedAt: string | null;
  scrapedAt: string;
}

export interface JobFilters {
  search?: string;
  source?: string;
  tags?: string[];
}

File: backend/schemas/job.py
"""Job response schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class JobOut(BaseModel):
    id: str
    title: str
    company: str
    company_id: str | None
    source_id: str | None
    location: str | None
    salary: str | None
    description: str | None
    apply_url: str | None
    sector: str | None
    stage: str | None
    tags: str | None
    is_new: bool
    is_remote: bool
    saved_at: datetime | None
    scraped_at: datetime

    model_config = ConfigDict(from_attributes=True)


class JobCountOut(BaseModel):
    count: int
